src/emitter.ts

```diff
@@ -1068,15 +1068,39 @@ export function emitWebIDl(webidl: Browser.WebIdl, flavor: Flavor) {
             return `[${types.join(", ")}]`;
         }
 
-        function emitIterableDeclarationMethods(subtypes: string[]) {
+        function emitIterableDeclarationMethods(i: Browser.Interface, subtypes: string[]) {
             let [keyType, valueType] = subtypes;
             if (!valueType) {
                 valueType = keyType;
                 keyType = "number";
             }
-            printer.printLine(`entries(): IterableIterator<[${keyType}, ${valueType}]>;`);
-            printer.printLine(`keys(): IterableIterator<${keyType}>;`);
-            printer.printLine(`values(): IterableIterator<${valueType}>;`);
+
+            const methods = [{
+                name: 'entries',
+                definition: `IterableIterator<[${keyType}, ${valueType}]>`,
+            }, {
+                name: 'values',
+                definition: `IterableIterator<${valueType}>`,
+            }, {
+                name: 'keys',
+                definition: `IterableIterator<${keyType}>`,
+            }];
+
+            const comments = i.iterator
+                && i.iterator.comments
+                && i.iterator.comments.comment;
+
+            methods.forEach((m) => {
+                if (comments && comments[m.name]) {
+                    printer.printLine(comments[m.name]);
+                }
+                printer.printLine(`${ m.name }(): ${ m.definition };`);
+            });
+
+            if (comments && comments['forEach']) {
+                printer.printLine(comments['forEach']);
+            }
+            emitIteratorForEach(i);
```

Comment:
not sure why `forEach` should be in this file and not in the normal DOM declarations..

this file is only separate because it has references to ES6-specific types like `Symbol.Iterator` and `IterableIterator`. `forEach` seems not to have any of these, and i would just treat it like a normal method.

Reply:
@mhegazy ok, will remove it